scylla/src/transport/cluster.rs

```diff
@@ -481,6 +458,17 @@ impl ClusterData {
     pub fn replica_locator(&self) -> &ReplicaLocator {
         &self.locator
     }
+
+    /// Returns iterator of working connections to all shards
+    pub(crate) fn iter_working_connections(
```

Comment:
`get_working_connections` returned an error when there were no working connections, by changing that some methods are going to panic.

Reply:
Do you mean the case when there are some nodes, but no working connections, or the case when there are no nodes?
In the first case, I can't see any possibility to trigger panic in both places that call this function.
I don't think that the second case is permitted by our invariant.

Reply:
If there are some nodes, but no working connections `iter_working_connections` will return an empty iterator.
```rust
        // Safety: there is at least one node in the cluster, and `Cluster::get_working_connections()`
        // returns at least one connection or an error, so there will be at least one result.
        let first_ok: Result<PreparedStatement, QueryError> =
            results.by_ref().find_or_first(Result::is_ok).unwrap();
```
[ref](https://github.com/scylladb/scylla-rust-driver/pull/780/files#diff-80ba8e49418bed2ad375905c49b6784a23c454eac059d99c3e18679a8525d9feR882)

I believe that:
- the safety comment is wrong here (`get_working_connections` was swapped for `iter_working_connections`)
- assumption about `iter_working_connections` returning at least one element is incorrect

Reply:
> If there are some nodes, but no working connections iter_working_connections will return an empty iterator.

This is incorrect. `iter_working_connections()` returns a `Result` for every node. It is either `Err` or `Ok`, but the iterator will definitely not be empty.

Reply:
Oops, you're right! I wrongly assumed what `flatten_ok` does in `iter_working_connections`.